sqs/src/main/scala/akka/stream/alpakka/sqs/impl/AutoBalancingSqsReceive.scala

```diff
@@ -0,0 +1,32 @@
+package akka.stream.alpakka.sqs.impl
+
+import akka.stream.FlowShape
+import akka.stream.scaladsl.{Broadcast, Flow, GraphDSL, Merge, Partition}
+import software.amazon.awssdk.services.sqs.SqsAsyncClient
+import software.amazon.awssdk.services.sqs.model.{Message, ReceiveMessageRequest, ReceiveMessageResponse}
+
+import scala.collection.JavaConverters._
+import scala.compat.java8.FutureConverters._
+
+class AutoBalancingSqsReceive(parallelism: Int)(implicit sqsClient: SqsAsyncClient) {
```

Comment:
hey @ennru 
I've put another way to solve the same problem.

This one uses routing to a reduced parallelism async stage instead of throttling, you might find this one cleaner.